index.js

```diff
@@ -163,7 +164,7 @@ function parseStyles(
 ) {
   var statements = parseStatements(result, styles)
 
-  return Promise.all(statements.map(function(stmt) {
+  return Promise.resolve(statements).then(promiseEach(function(stmt) {
```

Comment:
@MoOx I should have explained these changes better.

`map` runs the function down to `return resolveImportId()`. This starts a promise chain. `map` then goes on to the next item in `statements`, without waiting for `resolveImportId` to resolve. This causes concurrency. Depending on the order that the files are loaded, `state` is modified incorrectly. If `foo-second.css` loads first, `bar.css` is added to `state.importedFiles`. When `foo-first.css` is loaded, `bar.css` appears to be already loaded, however, it is in the incorrect order in the bundle.

`promiseEach` waits for `resolveImportId` to resolve before going to the next item in `statements`. promise-each is a standalone port of http://bluebirdjs.com/docs/api/promise.each.html.

This is a performance cut; but this is the quickest way to fix this. If someone can figure out a better deduping system that does not require serialized loading, great; I can't ATM.

Hope this is clear.